workload/src/client.h

```diff
@@ -1,27 +1,116 @@
-#ifndef __INCLUDE_WORKLOAD_CLIENT_H__
-#define __INCLUDE_WORKLOAD_CLIENT_H__
+#ifndef INCLUDE_WORKLOAD_CLIENT_H
+#define INCLUDE_WORKLOAD_CLIENT_H
 
-#include "workload.pb.h"
 #include "../../bundle/x509bundle/src/set.h"
 #include "../../svid/jwtsvid/src/svid.h"
-// #include "../../utils/src/util.h"
+#include "../../svid/x509svid/src/svid.h"
+#include "../../utils/src/util.h"
+#include "backoff.h"
+#include "watcher.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-x509bundle_Set* workloadapi_parseX509Bundles(
-    const X509SVIDResponse *rep, err_t *err);
-x509bundle_Bundle* workloadapi_parseX509Bundle(string_t id,
-                                            const byte *bundle_bytes,
-                                            const size_t len,
-                                            err_t *err);
-jwtsvid_SVID* workloadapi_parseJWTSVID(
-    const JWTSVIDResponse *resp, jwtsvid_Params *params, err_t *err);
-jwtbundle_Set* workloadapi_parseJWTBundles(
-    const JWTBundlesResponse *resp, err_t *err);
+/// pointer to gRPC construct, can't use those types in the header
+typedef void *workloadapi_Stub; // api stub
+typedef struct workloadapi_Watcher workloadapi_Watcher;
+
+typedef struct workloadapi_Client {
+    workloadapi_Stub stub;
+    bool ownsStub;
+    string_arr_t headers;
+    string_t address;
+    bool closed;
+    mtx_t closedMutex;
+    cnd_t closedCond;
```

Comment:
Replace "ownsStub", "closedMutex", "closedCond" for "owns_stub", "closed_mutex" and "closed_cond", respectively